In [7]:
import pandas as pd
pd.set_option('display.max_rows', 10000)

In [8]:
cd drive/Othercomputers/내 노트북/googleDrive/Wise-Saying

[Errno 2] No such file or directory: 'drive/Othercomputers/내 노트북/googleDrive/Wise-Saying'
/content/drive/Othercomputers/내 노트북/googleDrive/Wise-Saying


In [47]:
dataset = pd.read_csv("EN_dataset.csv")
dataset

author  \
0                   Marilyn Monroe   
1                William W. Purkey   
2                        Dr. Seuss   
3                   Elbert Hubbard   
4          Martin Luther King Jr.,   
...                            ...   
65785                Thomas Stark,   
65786                Thomas Stark,   
65787                Thomas Stark,   
65788                Thomas Stark,   
65789                Thomas Stark,   

                                                 content      tag  
0            “I'm selfish, impatient and a little ins...     love  
1            “You've gotta dance like there's nobody ...     love  
2            “You know you're in love when you can't ...     love  
3            “A friend is someone who knows all about...     love  
4            “Darkness cannot drive out darkness: onl...     love  
...                                                  ...      ...  
65785        “What is the eternal, necessary order? I...  science  
65786        “Light is the absolute condition for bot...  science  
65787        “Einstein understood all material frames...  science  
65788        “Light is not something different from t...  science  
65789        “Philosophy is dead. Only ontological ma...  science  

[65790 rows x 3 columns]

In [10]:
count = 0
for i in range(65790):
    if len(dataset['content'][i]) < 110:
        count += 1
print(count)

25931


### 콘텐츠 문장 길이가 110글자 이하인 것만 학습으로 이용
너무 긴 문장은 앱 내에서 출력하기 불편하다.
총 25931개의 데이터만 학습에 사용한다.

In [48]:
small_dataset = dataset
small_dataset['content_len'] = [0] * 65790
# 콘텐츠 길이 기준으로 자르기 위해 새로운 열 하나를 생성한다
small_dataset

author  \
0                   Marilyn Monroe   
1                William W. Purkey   
2                        Dr. Seuss   
3                   Elbert Hubbard   
4          Martin Luther King Jr.,   
...                            ...   
65785                Thomas Stark,   
65786                Thomas Stark,   
65787                Thomas Stark,   
65788                Thomas Stark,   
65789                Thomas Stark,   

                                                 content      tag  content_len  
0            “I'm selfish, impatient and a little ins...     love            0  
1            “You've gotta dance like there's nobody ...     love            0  
2            “You know you're in love when you can't ...     love            0  
3            “A friend is someone who knows all about...     love            0  
4            “Darkness cannot drive out darkness: onl...     love            0  
...                                                  ...      ...          ...  
65785        “What is the eternal, necessary order? I...  science            0  
65786        “Light is the absolute condition for bot...  science            0  
65787        “Einstein understood all material frames...  science            0  
65788        “Light is not something different from t...  science            0  
65789        “Philosophy is dead. Only ontological ma...  science            0  

[65790 rows x 4 columns]

In [49]:
for i in range(65790):
    small_dataset['content_len'][i] = len(small_dataset['content'][i])
small_dataset.tail()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


author                                            content  \
65785      Thomas Stark,        “What is the eternal, necessary order? I...   
65786      Thomas Stark,        “Light is the absolute condition for bot...   
65787      Thomas Stark,        “Einstein understood all material frames...   
65788      Thomas Stark,        “Light is not something different from t...   
65789      Thomas Stark,        “Philosophy is dead. Only ontological ma...   

           tag  content_len  
65785  science          461  
65786  science          464  
65787  science          439  
65788  science          686  
65789  science          345

In [50]:
small_dataset = small_dataset.drop(small_dataset[small_dataset['content_len'] > 110].index)
small_dataset.reset_index(inplace=True)
small_dataset

index                author  \
0          2             Dr. Seuss   
1          3        Elbert Hubbard   
2          5      Stephen Chbosky,   
3          6           Andre Gide,   
4          8           John Green,   
...      ...                   ...   
26273  65748      Nouhayla Elmerss   
26274  65757           Mary Beard,   
26275  65760       Galileo Galilei   
26276  65769         Tom B. Night,   
26277  65778         Thomas Stark,   

                                                 content      tag  content_len  
0            “You know you're in love when you can't ...     love          110  
1            “A friend is someone who knows all about...     love           72  
2              “We accept the love we think we deserve.”     love           47  
3            “It is better to be hated for what you a...     love           88  
4            “As he read, I fell in love the way you ...     love           89  
...                                                  ...      ...          ...  
26273                      “You can't dull my sparkle ✨”  science           35  
26274        “Denn bei historischer Forschung geht es...  science           94  
26275        “Who would set a limit to the mind? Who ...  science          103  
26276        “With a fixed amount of matter and energ...  science          110  
26277        “Mathematics is the ultimate superhero –...  science           93  

[26278 rows x 5 columns]

### 특이 값 제거

In [53]:
small_dataset = small_dataset.drop([59,
554,
706,
894,
966,
1025,
1194,
1111,
1290,
1385,
1335,
1315,
1452,
1622,
1699,
1856,
1870,
1985,
2148,
2429,
2562,
2855,
2805,
2958,
2935,
3072,
3030,
3023,
3193,
3333,
3453,
5174,
5227,
5379,
5380,
5422,
5509,
5547,
5555,
5581,
5585,
5656,
5754,
5729,
5885,
5847,
5820,
5902,
5911,
5913,
5914,
5915,
6095,
6186,
6286,
6214,
6210,
6362,
6360,
6341,
6461,
6435,
6728,
7299,
7536,
7648,
7748,
4980,
8010,
8101,
8260,
8319,
8309,
8464,
8435,
8545,
8500,
8711,
8836,
8948,
8904,
9098,
9083,
9006,
9262,
9241,
9202,
9775,
10053,
10013,
10103,
10323,
10591,
10537,
10534,
10531,
10530,
10631,
10642,
10916,
11036,
11024,
11196,
11285,
11364,
11495,
11556,
11690,
11772,
11868,
11950,
12068,
12143,
12120,
12117,
12342,
12341,
12340,
12309,
12695,
12621,
12854,
13072,
13059,
13152,
13100,
13246,
13313,
13493,
13494,
13495,
13496,
14092,
14192,
14419,
14524,
14526,
14529,
14533,
14534,
14543,
15099,
15397,
15375,
16461,
16475,
16476,
16482,
16587,
16627,
16756,
17083,
17038,
17196,
17290,
17270,
17263,
17374,
17593,
17547,
17856,
17914,
19338,
19330,
19313,
19450,
19472,
19466,
19436,
19431,
19412,
19573,
19571,
19550,
19527,
19517,
19502,
19695,
19671,
19670,
19669,
19667,
19666,
19665,
19664,
19663,
19662,
19661,
19660,
19659,
19655,
19620,
19619,
19618,
19617,
19615,
19922,
19922,19924])

### content 외의 열 제거

In [55]:
small_dataset = small_dataset.drop(columns = ['index','author', 'tag', 'content_len'])  #여러개 열을 삭제할 때.
small_dataset

content
0            “You know you're in love when you can't ...
1            “A friend is someone who knows all about...
2              “We accept the love we think we deserve.”
3            “It is better to be hated for what you a...
4            “As he read, I fell in love the way you ...
...                                                  ...
26273                      “You can't dull my sparkle ✨”
26274        “Denn bei historischer Forschung geht es...
26275        “Who would set a limit to the mind? Who ...
26276        “With a fixed amount of matter and energ...
26277        “Mathematics is the ultimate superhero –...

[26080 rows x 1 columns]

In [57]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the "../input/" directory.
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory
import tensorflow as tf
import matplotlib.pyplot as plt
from tqdm import tqdm

!pip install mitdeeplearning
import mitdeeplearning as mdl

import os

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 2.1 MB 7.3 MB/s 
  Created wheel for mitdeeplearning: filename=mitdeeplearning-0.2.0-py3-none-any.whl size=2115442 sha256=01454b02fc18c48c26e35caa26ef9dbb3930a22fcde57ab43c98c74225377055
  Stored in directory: /root/.cache/pip/wheels/9a/b9/4f/99b7c8c5c75355550b83e1fcfc02956fb40c35eb01e2262877
Successfully built mitdeeplearning


In [58]:
small_dataset.to_csv("EN_train.csv", index=False)